packages/backend-tasks/src/tasks/LocalTaskWorker.test.ts

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { getVoidLogger } from '@backstage/backend-common';
+import { AbortController } from 'node-abort-controller';
+import { LocalTaskWorker } from './LocalTaskWorker';
+
+describe('LocalTaskWorker', () => {
+  const logger = getVoidLogger();
+
+  it('runs the happy path (with iso duration) and handles cancellation', async () => {
+    const fn = jest.fn();
+    const controller = new AbortController();
+
+    const worker = new LocalTaskWorker('a', fn, logger);
+    worker.start(
+      {
+        version: 2,
+        initialDelayDuration: 'PT0.2S',
+        cadence: 'PT0.2S',
+        timeoutAfterDuration: 'PT1S',
+      },
+      { signal: controller.signal },
+    );
+
+    // TODO(freben): Rewrite to fake timers - tried, but it wouldn't work
```

Comment:
I felt dumb but I really for real tried and failed with both modern and legacy. Thing is, cron is wall clock based and needs system time to be set (implying modern timers), and that didn't jive with ALSO pumping micro tasks. And any await of a small setInterval in the test itself
will also "hang" the test since THAT isn't pumping. Went back and forth but frustratedly went back to regular blocking tests instead :/ Will have to revisit these. Their margins are rather large at least, so one has to get pretty  damn swamped to get flakiness.